src/gd_heif.c

```diff
@@ -0,0 +1,839 @@
+
+    <gdImageCreateFromHeif> is called to load truecolor images from
+    HEIF format files. Invoke <gdImageCreateFromHeif> with an
+    already opened pointer to a file containing the desired
+    image. <gdImageCreateFromHeif> returns a <gdImagePtr> to the new
+    truecolor image, or NULL if unable to load the image (most often
+    because the file is corrupt or does not contain a HEIF
+    image). <gdImageCreateFromHeif> does not close the file.
+
+    You can inspect the sx and sy members of the image to determine
+    its size. The image must eventually be destroyed using
+    <gdImageDestroy>.
+
+    *The returned image is always a truecolor image.*
+
+  Variants:
+
+    <gdImageCreateFromAvif> AVIF method counterpart.
+
+  Parameters:
+
+    infile - The input FILE pointer.
+
+  Returns:
+
+    A pointer to the new *truecolor* image.  This will need to be
+    destroyed with <gdImageDestroy> once it is no longer needed.
+
+    On error, returns NULL.
+*/
+BGD_DECLARE(gdImagePtr) gdImageCreateFromHeif (FILE * inFile)
+{
+	gdImagePtr im;
+	gdIOCtx *in = gdNewFileCtx(inFile);
+	if (!in) {
+		return 0;
+	}
+	im = gdImageCreateFromHeifCtx(in);
+	in->gd_free(in);
+
+	return im;
+}
+
+/*
+  Function: gdImageCreateFromHeifPtr
+
+    See <gdImageCreateFromHeif>.
+
+  Parameters:
+
+    size            - size of HEIF data in bytes.
+    data            - pointer to HEIF data.
+*/
+BGD_DECLARE(gdImagePtr) gdImageCreateFromHeifPtr (int size, void *data)
+{
+	gdImagePtr im;
+	gdIOCtx *in = gdNewDynamicCtxEx(size, data, 0);
+	if (!in)
+		return 0;
+	im = gdImageCreateFromHeifCtx(in);
+	in->gd_free(in);
+	return im;
+}
+
+static int _gdHeifCheckBrand (unsigned char *magic, gd_heif_brand expected_brand)
+{
+	if (memcmp(magic + 4, "ftyp", 4) != 0)
+		return GD_FALSE;
+	if (memcmp(magic + 8, "avif", 4) == 0 && expected_brand & GD_HEIF_BRAND_AVIF)
+		return GD_TRUE;
+	if (memcmp(magic + 8, "heic", 4) == 0 && expected_brand & GD_HEIF_BRAND_HEIC)
+		return GD_TRUE;
+	if (memcmp(magic + 8, "heix", 4) == 0 && expected_brand & GD_HEIF_BRAND_HEIX)
+		return GD_TRUE;
+	if (memcmp(magic + 8, "mif1", 4) == 0 && expected_brand & GD_HEIF_BRAND_MIF1)
+		return GD_TRUE;
+
+	return GD_FALSE;
+}
+
+static gdImagePtr _gdImageCreateFromHeifCtx (gdIOCtx * infile, gd_heif_brand expected_brand)
+{
+	struct heif_context *heif_ctx;
+	struct heif_decoding_options *heif_dec_opts;
+	struct heif_image_handle *heif_imhandle;
+	struct heif_image *heif_im;
+	struct heif_error err;
+	int    width, height;
+	uint8_t   *filedata = NULL;
+	uint8_t    *rgba = NULL;
+	unsigned char   *read, *temp, *magic;
+	size_t size = 0, n;
+	gdImagePtr im;
+	int x, y;
+	uint8_t *p;
+
+	magic = gdMalloc(GD_HEIF_HEADER);
+	gdGetBuf(magic, GD_HEIF_HEADER, infile);
+	if (!_gdHeifCheckBrand(magic, expected_brand)) {
+		gd_error("gd-heif incorrect type of file\n");
+		gdFree(magic);
+		return NULL;
+	}
+	gdFree(magic);
+	gdSeek(infile, 0);
+
+	do {
+		temp = gdRealloc(filedata, size+GD_HEIF_ALLOC_STEP);
```

Comment:
use spaces around binary operators like `+` and `>` and `!=`

this comes up multiple times in this file